test/compilable/dtoh_enum_cpp98.d

```diff
@@ -233,7 +233,7 @@ extern(C++) enum c = [2: 3];
 extern(C) void foo();
 extern(C++) enum d = &foo;
 
-immutable bool e_b;
+__gshared immutable bool e_b;
```

Comment:
Immutable should be shared anyway though?

Reply:
If that was true, there wouldn't be an [internal error](https://run.dlang.io/is/co4la8).

Reply:
In which case it shouldn't be emitted at all, no?

Reply:
This is what I'm asking @MoonlightSentinel.  TLS symbols shouldn't be emitted, but at the same time such tests should be in fail_compilation in order not to trigger the ICE in cppmangle.

Reply:
The missing `__gshared` was not intentional.
The glue layer shouldn't reject `immutable` variables but that's another discussion.

Reply:
Further to the note of `immutable` being implicitly `shared`.  D `shared` does not map to C++ either.